### ISSUE TITLE
Signed assembly is empty

### ISSUE THREAD

**Issue by ejsmith** [atifaziz/NCrontab#48]
In the new 3.3.1 release the signed version of the package is empty. The non-signed version works fine.

**Comment by atifaziz** [comment]
I'm not seeing the same thing. If I install the signed package:

```
$ NuGet install ncrontab.signed -Out $TMPDIR`
Feeds used:
  https://api.nuget.org/v3/index.json

Installing package 'ncrontab.signed' to '/var/folders/y5/0kw6v1lj2cx3bjqm0b003m480000gn/T/'.
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/ncrontab.signed/index.json
Package "ncrontab.signed.3.3.1" is already installed.
```

and look at its contents:

```
$ tree $TMPDIR/NCrontab.Signed.3.3.1 
/var/folders/y5/0kw6v1lj2cx3bjqm0b003m480000gn/T//NCrontab.Signed.3.3.1
├── COPYING.txt
├── NCrontab.Signed.3.3.1.nupkg
└── lib
    ├── net35
    │   ├── NCrontab.Signed.dll
    │   └── NCrontab.Signed.pdb
    ├── netstandard1.0
    │   ├── NCrontab.Signed.dll
    │   └── NCrontab.Signed.pdb
    └── netstandard2.0
        ├── NCrontab.Signed.dll
        └── NCrontab.Signed.pdb

4 directories, 8 files
```

then it doesn't seem to be empty.

**Comment by ejsmith** [comment]
Yes, it has assemblies, but there are no types in the assemblies. They are empty.

**Comment by ejsmith** [comment]
I am specifically on netstandard2.0 but I think it was the same for all of them.

**Comment by atifaziz** [comment]
Wow, you're right! I ran [`ilspycmd`](https://www.nuget.org/packages/ilspycmd/) to decompile the DLL and it has no types defined in it whatsoever:

```
$ ilspycmd lib/netstandard2.0/NCrontab.Signed.dll
using System.Diagnostics;
using System.Reflection;
using System.Resources;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using System.Runtime.Versioning;

[assembly: CompilationRelaxations(8)]
[assembly: RuntimeCompatibility(WrapNonExceptionThrows = true)]
[assembly: Debuggable(DebuggableAttribute.DebuggingModes.IgnoreSymbolStoreSequencePoints)]
[assembly: AssemblyProduct("NCrontab")]
[assembly: AssemblyCopyright("Copyright © 2008, Atif Aziz. All rights reserved. Portions Copyright © 2001 The OpenSymphony Group. All rights reserved.")]
[assembly: AssemblyCompany("")]
[assembly: AssemblyTrademark("")]
[assembly: AssemblyConfiguration("RELEASE")]
[assembly: ComVisible(false)]
[assembly: TargetFramework(".NETStandard,Version=v2.0", FrameworkDisplayName = "")]
[assembly: AssemblyInformationalVersion("3.3.1+b6659e45d60de554c5619d7f35f39ad1114b6542")]
[assembly: AssemblyTitle("NCrontab (Signed)")]
[assembly: NeutralResourcesLanguage("en-US")]
[assembly: AssemblyVersion("0.0.0.0")]
```

How did that happened? 😕

**Comment by ejsmith** [comment]
LOL yeah, good question. 😀 I’m sure it will be something dumb like it always is. I was going to take a look yesterday, but I had already worked the entire day.

**Comment by atifaziz** [comment]
Ah, it seems that in d47ad6cd32e68eb6753c8d904299b67982eeb960, I blew away one line too many that included all the C# source files (`*.cs`):

```diff
diff --git a/NCrontab.Signed/NCrontab.Signed.csproj b/NCrontab.Signed/NCrontab.Signed.csproj
index dbddbcb..70f8cfd 100644
--- a/NCrontab.Signed/NCrontab.Signed.csproj
+++ b/NCrontab.Signed/NCrontab.Signed.csproj
@@ -26,7 +26,7 @@ ﻿<Project Sdk="Microsoft.NET.Sdk">
 3.0: Replaces uses of Converter with Func and therefore requires .NET Framework 3.5 as a minimum.
 </PackageReleaseNotes>
     <PackageProjectUrl>https://github.com/atifaziz/NCrontab</PackageProjectUrl>
-    <PackageLicenseUrl>http://www.apache.org/licenses/LICENSE-2.0</PackageLicenseUrl>
+    <PackageLicenseFile>COPYING.txt</PackageLicenseFile>
     <GenerateAssemblyDescriptionAttribute>false</GenerateAssemblyDescriptionAttribute>
     <GenerateAssemblyVersionAttribute>false</GenerateAssemblyVersionAttribute>
     <GenerateAssemblyFileVersionAttribute>false</GenerateAssemblyFileVersionAttribute>
@@ -38,10 +38,7 @@ ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <ItemGroup>
     <Compile Include="..\SolutionInfo.cs" />
-    <Compile Include="..\NCrontab\*.cs" Exclude="bin\**;obj\**;**\*.xproj;packages\**" />
-    <None Update="..\COPYING.txt;..\lic\OpenSymphony.txt">
-      <Pack>true</Pack>
-    </None>
+    <None Include="..\COPYING.txt" Pack="true" PackagePath="$(PackageLicenseFile)" />
   </ItemGroup>
 
   <!--
```

> I’m sure it will be something dumb like it always is.

And so sure enough it was. 🤦‍♂️ I think there ought to be a compiler warning for an assembly emitted without any types. 😛

I need to find a smarter way to keep `NCrontab.csproj` and `NCrontab.Signed.csproj` in sync rather than doing it by hand but I am no MSBuild expert.

**Comment by ejsmith** [comment]
The reasons for not signing your assembly used to be because it made binding redirects a nightmare, but that isn’t the same any more. You should read through this site:

https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming

**Comment by atifaziz** [comment]
> it made binding redirects a nightmare, but that isn’t the same any more.

Trouble is that it's still a nightmare for .NET Framework:

> Fortunately, this issue is isolated to the .NET Framework. .NET Core, Xamarin, UWP, and most other .NET implementations don't have strict assembly loading and removes the main downside of strong naming.

**Comment by atifaziz** [comment]
@ejsmith Could you test [NCrontab.Signed 3.3.2-ci-20190217T2148](https://www.myget.org/feed/raboof/package/nuget/NCrontab.Signed/3.3.2-ci-20190217T2148) and confirm it's good to go?. You can install it using the following command-line:

    NuGet install NCrontab.Signed -Version 3.3.2-ci-20190217T2148 -Source https://www.myget.org/F/raboof/api/v3/index.json

**Comment by ejsmith** [comment]
Right, but even that has gotten better and also they recommend you don't increment the assembly version unless you do a major breaking release which would be pretty much never with a simple package like this one.

> > it made binding redirects a nightmare, but that isn’t the same any more.
> 
> Trouble is that it's still a nightmare for .NET Framework:
> 
> > Fortunately, this issue is isolated to the .NET Framework. .NET Core, Xamarin, UWP, and most other .NET implementations don't have strict assembly loading and removes the main downside of strong naming.

Right, but even that has gotten better and also they recommend you don't increment the assembly version unless you do a major breaking release which would be pretty much never with a simple package like this one.

**Comment by atifaziz** [comment]
> Right, but even that has gotten better and also they recommend you don't increment the assembly version unless you do a major breaking release which would be pretty much never with a simple package like this one.

I'll consider that for the next release.

**Comment by atifaziz** [comment]
The fix is now published with [release 3.3.2](https://github.com/atifaziz/NCrontab/releases/tag/v3.3.2-sn) and package version 3.3.1 for [ncrontab.signed](https://www.nuget.org/packages/ncrontab.signed/) has been unlisted.
